app/user_prompt_support/user_prompt_manager.py

```diff
@@ -50,6 +50,15 @@ def __init__(self, prompt: PromptRequest, message_id: int) -> None:
     def as_dictionary(self) -> Dict[MessageKeysEnum, Any]:
         prompt_dict = self.prompt.dict()
         prompt_dict[MESSAGE_ID_KEY] = self.message_id
+        message_type = MessageTypeEnum.PROMPT_REQUEST
+
+        # Check if options exist
+        if "options" in prompt_dict:
+            if prompt_dict["options"]:
+                message_type = MessageTypeEnum.OPTIONS_REQUEST
+            else:
+                message_type = MessageTypeEnum.MESSAGE_REQUEST
+                
         message_dict = {
             MessageKeysEnum.TYPE: MessageTypeEnum.PROMPT_REQUEST,
             MessageKeysEnum.PAYLOAD: prompt_dict,
```

Comment:
As the Flake check alerted, we need to use the `message_type` created:
```suggestion
        message_dict = {
            MessageKeysEnum.TYPE: message_type,
            MessageKeysEnum.PAYLOAD: prompt_dict,
```